From ad14b4cc782bb1575de4712b0baee5640dff5bac Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 19 Aug 2014 19:30:49 -0400
Subject: [scsi] qla4xxx: Check for correct return status

Message-id: <1408476658-5056-12-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88344
O-Subject: [PATCH RHEL 7.1 e-stor 11/20] qla4xxx: Check for correct return status
Bugzilla: 1089349
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Vikas Chaudhary <vikas.chaudhary@qlogic.com>

scsi commit id 9c4f8d92c23de47db080d1a33addea30751f6122

Check for correct return status in function -
qla4_8xxx_minidump_pex_dma_read

Signed-off-by: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Reviewed-by: Mike Christie <michaelc@cs.wisc.edu>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla4xxx/ql4_nx.c b/drivers/scsi/qla4xxx/ql4_nx.c
index 8a56209..9dbdb4b 100644
--- a/drivers/scsi/qla4xxx/ql4_nx.c
+++ b/drivers/scsi/qla4xxx/ql4_nx.c
@@ -2105,7 +2105,7 @@ static int qla4_8xxx_minidump_pex_dma_read(struct scsi_qla_host *ha,
          (uint64_t)m_hdr->desc_card_addr,
          (uint32_t *)&dma_desc,
          (sizeof(struct qla4_83xx_pex_dma_descriptor)/16));
-  if (rval == -1) {
+  if (rval != QLA_SUCCESS) {
    ql4_printk(KERN_INFO, ha,
        "%s: Error writing rdmem-dma-init to MS !!!\n",
        __func__);
-- 
1.7.1